### PR TITLE
[tool] Removes deprecated --web-renderer parameter.

### PR DESCRIPTION
**Pull request by ditman**
Removes the `--web-renderer` option.

Flutter tool users won't be able to select their web renderer.

In order to not drop test coverage, tests are still run with the HTML Renderer, until its code is fully removed from the engine.

Part of https://github.com/flutter/flutter/issues/145954

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Comment by ditman** [comment]
Publishing this PR to have CI take a look. I fully expect this to fail.

0b0d048c-fd01-4136-a025-d4983c8ab082_text

**Comment by ditman** [comment]
The problem was that I was not using the correct renderer on flutter run/drive. Now I've corrected this ([4a2ae1d](https://github.com/flutter/flutter/pull/159314/commits/4a2ae1d35ca50c248b38868898182a36d5a4c0c5)), and tests should pass, unless I broke some by changing the errors. 🤞

**Comment by ditman** [comment]
In test_6, it seems the runner is doing:

> /work/flutter/flutter/bin/cache/flutter_web_sdk/ --incremental --target=dartdevc --experimental-emit-debug-metadata
-DFLUTTER_WEB_AUTO_DETECT=false **-DFLUTTER_WEB_USE_SKIA=false** -DFLUTTER_WEB_USE_SKWASM=false -DDART_HHH_BOT=false
**-DFLUTTER_WEB_USE_SKIA=true**

That's probably not good :)

**Comment by ditman** [comment]
All right, down to 1 failing test!

File: packages/flutter_tools/test/web.shard/hot_reload_web_test.dart [hunk 1]
```diff
@@ -48,7 +48,7 @@ Future<void> _testProject(HotReloadProject project, {String name = 'Default'}) a
 
   testWithoutContext('$testName: hot restart works without error', () async {
     flutter.stdout.listen(printOnFailure);
-    await flutter.run(chrome: true, additionalCommandArgs: <String>['--verbose', '--web-renderer=html']);
+    await flutter.run(chrome: true, additionalCommandArgs: <String>['--verbose']);
```
Comment (ditman) [inline packages/flutter_tools/test/web.shard/hot_reload_web_test.dart:51]: I'm going to restore the `html` renderer in these tests, to see if they cure the broken test, but that's definitely not a permanent fix :/

**Review by yjbanov — APPROVED** [review]
lgtm

**Comment by ditman** [comment]
All is green, I'll set this to `autosubmit` next week.

**Pull request merged**